modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/java/com/liferay/osb/faro/web/internal/util/PreferencesControllerUtil.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.osb.faro.web.internal.util;
+
+import com.liferay.osb.faro.model.FaroPreferences;
+import com.liferay.osb.faro.service.FaroPreferencesLocalService;
+import com.liferay.osb.faro.web.internal.constants.FaroPreferencesConstants;
+import com.liferay.osb.faro.web.internal.model.preferences.WorkspacePreferences;
+import com.liferay.portal.kernel.util.StringUtil;
+
+/**
+ * @author Joao Victor Alves
+ */
+public class PreferencesControllerUtil {
+
+	public static long getOwnerId(long groupId, String scope, long userId)
+		throws Exception {
+
+		if (StringUtil.equals(scope, FaroPreferencesConstants.SCOPE_GROUP)) {
+			return groupId;
+		}
+		else if (StringUtil.equals(
+					scope, FaroPreferencesConstants.SCOPE_USER)) {
+
+			return userId;
+		}
+
+		throw new Exception("Invalid scope " + scope);
+	}
+
+	public static WorkspacePreferences getWorkspacePreferences(
+			FaroPreferencesLocalService faroPreferencesLocalService,
+			long groupId, long ownerId)
+		throws Exception {
+
+		FaroPreferences faroPreferences =
```

Comment:
Hey @regiCesar21, pass the `FaroPreferences faroPreferences` as a method parameter instead.